### ISSUE TITLE
Chat with onnxruntime-web fails when i deployed on my machine

### ISSUE THREAD

**Issue by salimngit** [microsoft/onnxruntime-inference-examples#438]
Chat with onnxruntime-web example deployed with 

"onnxruntime-web": "1.19.0-dev.20240509-69cfcba38a"
and associated wasm files from https://cdn.jsdelivr.net/npm/onnxruntime-web@1.19.0-dev.20240509-69cfcba38a/dist/

live demo https://guschmue.github.io/ort-webgpu/chat/index.html works just fine.

When i run locally, i receive the below error

<html>
<body>
<div aria-expanded="true" tabindex="-1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 0 0 auto; color: rgb(27, 27, 27); font-family: consolas, &quot;lucida console&quot;, &quot;courier new&quot;, monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(211, 227, 253); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span class="source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; line-height: 1.2;"><span class="console-message-text" style="box-sizing: border-box; min-width: 0px; min-height: 0px; color: rgb(225, 0, 0);"><span class="object-value-error source-code" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--source-code-font-family); white-space: pre-wrap; font-size: var(--source-code-font-size) !important; line-height: 1.2;"><div style="box-sizing: border-box; min-width: 0px; min-height: 0px;">Error: found infinitive in logits
<span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">    at LLM.argmax (<button class="devtools-link text-button link-style" jslog="Link; context: script-location; track: click" role="link" tabindex="-1" title="webpack://localchat/llm.js:152:1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: 18px; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none !important; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;">llm.js:152:1</button>)
</span><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">    at LLM.generate (<button class="devtools-link text-button link-style" jslog="Link; context: script-location; track: click" role="link" tabindex="-1" title="webpack://localchat/llm.js:210:1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: 18px; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none !important; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;">llm.js:210:1</button>)
</span><span style="box-sizing: border-box; min-width: 0px; min-height: 0px;">    at async Query (<button class="devtools-link text-button link-style" jslog="Link; context: script-location; track: click" role="link" tabindex="-1" title="webpack://localchat/main.js:233:1" style="box-sizing: border-box; min-width: 0px; min-height: 0px; font: inherit; margin: 0px; height: 18px; border: none; border-radius: 2px; padding: 0px !important; color: var(--sys-color-primary); background: none !important; flex: 0 0 auto; white-space: nowrap; text-decoration: underline; outline-offset: 2px; outline: none; cursor: pointer; word-break: break-all;">main.js:233:1</button>)</span></div></span></span></span></div><div class="" role="group" style="box-sizing: border-box; min-width: 0px; min-height: 0px; flex: 0 0 auto; color: rgb(27, 27, 27); font-family: consolas, &quot;lucida console&quot;, &quot;courier new&quot;, monospace; font-size: 12px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; white-space: normal; background-color: rgb(211, 227, 253); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><span class="monospace stack-preview-container width-constrained" style="display: inline-block; --monospace-font-size: 12px; --monospace-font-family: consolas, lucida console, courier new, monospace; --source-code-font-size: 12px; --source-code-font-family: consolas, lucida console, courier new, monospace; width: 639px; box-sizing: border-box; min-width: 0px; min-height: 0px; font-family: var(--monospace-font-family); font-size: var(--monospace-font-size) !important;">
  | (anonymous) | @ | main.js:141
-- | -- | -- | --
  | Promise.catch (async) |   |  
  | submitRequest | @ | main.js:140
  | (anonymous) | @ | main.js:167

</span></div>
</body>
</html>

**Comment by fosteman** [comment]
It looks like you're encountering an error related to the presence of infinite values in the logits during the inference process with `onnxruntime-web`. This can happen due to various reasons, including model issues, input data anomalies, or bugs in the runtime.

Here are a few steps you can take to troubleshoot and resolve this issue:

1. **Check Input Data:**
   - Ensure that the input data fed to the model is properly preprocessed and normalized. Incorrect input data can lead to invalid computations resulting in infinite values.

2. **Model Integrity:**
   - Verify that the model file is not corrupted. Re-download the model if necessary and ensure that it is correctly loaded in your code.

3. **Runtime Version:**
   - Make sure you are using the correct version of `onnxruntime-web` and associated WASM files. Since you mentioned using a specific development version, confirm that the paths to the WASM files are correct and they are accessible.

4. **Check for Known Issues:**
   - Look for any known issues or bugs reported with the version you are using. Sometimes, development versions might have unresolved bugs that could be causing the problem.

5. **Error Handling:**
   - Add additional error handling in your code to catch and log the specific values causing the issue. This can help pinpoint the exact cause of the infinite values.

Here is an example of how you might add error handling to check for infinite values in logits:

```javascript
async function generateResponse(input) {
    try {
        const logits = await LLM.generate(input);
        
        // Check for infinite values in logits
        if (logits.some(value => !isFinite(value))) {
            throw new Error("Found infinite value in logits");
        }
        
        const response = LLM.argmax(logits);
        return response;
    } catch (error) {
        console.error("Error in generateResponse:", error);
        // Handle the error appropriately
    }
}

async function submitRequest(input) {
    try {
        const response = await generateResponse(input);
        // Process the response
    } catch (error) {
        console.error("Error in submitRequest:", error);
        // Handle the error appropriately
    }
}
```

### Verify Local Setup:
- Ensure all dependencies are correctly installed.
- Check the console logs for any additional warnings or errors that might give more insight into the problem.

### Compare with Working Demo:
- Compare your local setup with the live demo setup to identify any configuration differences. Ensure all settings, versions, and paths match the working demo.

### Debugging Tips:
- Log intermediate values to understand where the infinite values are introduced.
- Simplify your input and model to isolate the issue, starting with basic inputs and gradually increasing complexity.

By following these steps, you should be able to identify and resolve the issue with infinite values in the logits during model inference.
